### PR TITLE
implement unique op for fake tensor

### PR DESCRIPTION
**Pull request by cascade812**
implement unique op for fake tensor

### CODE REVIEW

**Comment by guijuzhang** [comment]
@eellison could you help me review this PR, thanks!

File: torch/_subclasses/fake_impls.py [hunk 1]
```diff
@@ -329,6 +329,24 @@ def nonzero(fake_mode, func, arg):
...
+        # Without symints/symfloats, cannot handle this
+        raise DynamicOutputShapeException(func)
+    
+    uni_cnt = fake_mode.shape_env.create_unbacked_symint()
+    # Avoid importing sympy at a module level
+    from torch.fx.experimental.symbolic_shapes import (
+        _constrain_range_for_size
+    )
+    _constrain_range_for_size(uni_cnt, min=1, max=arg.numel())
+    return arg.new_empty(uni_cnt), arg.new_empty(arg.size(), dtype=torch.int64), arg.new_empty(uni_cnt, dtype=torch.int64)
```
Comment (eellison) [inline torch/_subclasses/fake_impls.py:347]: The shape of the last two returned tensors depend on `return_inverse` and `return_counts`

**Review by eellison — CHANGES_REQUESTED** [review]
Looks good but we're not correctly setting the shape of the last two tensors

File: torch/_subclasses/fake_impls.py (hunk 1 above)
Comment (guijuzhang) [reply torch/_subclasses/fake_impls.py:347]: @eellison  thanks for your review!
`kwargs `doesn't carry `return_inverse` and `return_counts`, is this expected?
Also when I test with `torch.unique(x, sorted=True, return_inverse=True, return_counts=False)`, it throws errror "unique() takes 3 positional arguments but 5 were given". But `torch.unique(x, sorted=True, return_inverse=False, return_counts=False)` works good at the same time. do you happen to know why? Seems dispatch logic is having some problem to me.

Comment (guijuzhang) [reply torch/_subclasses/fake_impls.py:347]: @eellison thanks for your review. The caller of _unique2 (File "/root/pytorch/torch/functional.py", line 910, in _unique_impl) requires three results regardless the value of return_inverse and return_counts
```
output, inverse_indices, counts = torch._unique2(
            input,
            sorted=sorted,
            return_inverse=return_inverse,
            return_counts=return_counts,
        )
```

**Comment by guijuzhang** [comment]
@eellison could you help take another look? thanks!

**Comment by guijuzhang** [comment]
@eellison  would you pls take a another look? thanks a lot.

File: test/test_fake_tensor.py [hunk 2]
```diff
@@ -554,8 +554,19 @@ def test_cuda_lstm(self):
...
-
             self.assertRaises(DynamicOutputShapeException, lambda: torch.nonzero(x))
+            self.assertRaises(DynamicOutputShapeException, lambda: torch.unique(x))
+
+        # unique()
+        shape_env = ShapeEnv()
+        with FakeTensorMode(shape_env=shape_env):
+            x = torch.rand([10, 10])
+            y = torch.unique(x)
+            self.assertEqual(type(y.shape[0]), torch.SymInt)
```
Comment (eellison) [inline test/test_fake_tensor.py:565]: Please add a couple tests using `torch.ops.aten._unique2.default` and `return_counts` and `return_inverse`

**Review by eellison — COMMENTED** [review]
@guijuzhang you can yes the aten.ops invocation directly to see that the output shape is different depending on the kwargs

File: test/test_fake_tensor.py (hunk 2 above)
Comment (guijuzhang) [reply test/test_fake_tensor.py:565]: added test cases as requested.

**Pull request closed without merging**